### PR TITLE
Pre-flight PlanMyTrip performance issues bug fix

### PR DESCRIPTION
**Pull request by dogcomplex**
# Overview

This PR includes the following proposed change(s):

- Fixes critical bug causing multiple updates on every single pouchdb insert, and app freezing
- Fixes duplicate docs bug from earlier PR
- Bumps up cached activities/POIs support to 5k objects, with relatively speedy performance
- Refactors TripDataControls somewhat

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Refactoring / Documentation

## How Has This Been Tested?

Locally

## Screenshots

2021-02-25 04_24_00-InvasivesBC - Map
2021-02-25 04_22_44-InvasivesBC - Plan
2021-02-25 04_24_31-InvasivesBC - Map

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by micheal-w-wells** [comment]
Got a conflict to resolve here champ - if i cant get to it first I'll do my best but not having written it might be better if you do. @dogcomplex

File: app/src/components/trip/TripDataControls.tsx [hunk 1]
```diff
@@ -36,12 +36,65 @@ export const TripDataControls: React.FC = (props) => {
...
+        return upserts[id];
+      })
+      .map((doc) => upserts[doc.id](doc));
+
+    const newDocs = Object.keys(newUpserts)
+      .filter((id) => newUpserts[id] !== undefined)
+      .map((id) => upserts[id]());
+
+    const resultDocs = [
+      // ...modifiedDocs,
```
Comment (sdevalapurkar) [inline app/src/components/trip/TripDataControls.tsx:59]: nit: remove this line?

File: app/src/components/trip/TripDataControls.tsx [hunk 2]
```diff
@@ -36,12 +36,65 @@ export const TripDataControls: React.FC = (props) => {
   const [trip, setTrip] = useState(null);
   const [fetching, setFetching] = useState(false);
 
-  const getTrip = useCallback(async () => {
-    let docs = await databaseContext.database.find({ selector: { _id: 'trip' } });
+  const bulkUpsert = async (upserts) => {
+    let allDocsFetch = await databaseContext.database.allDocs({ include_docs: true });
+    let allDocs = allDocsFetch?.rows ? allDocsFetch.rows : [];
```
Comment (sdevalapurkar) [inline app/src/components/trip/TripDataControls.tsx:41]: nit: both can be consts?

**Review by sdevalapurkar — APPROVED** [review]
LGTM, just a few minor nits

**Pull request merged**